utils/fixed_decimal/src/decimal.rs

```diff
@@ -1819,8 +1819,10 @@ impl FromStr for FixedDecimal {
         // The string without the exponent (or sign)
         // We do the bulk of the calculation on this string,
         // and extract the exponent at the end
-        #[allow(clippy::indexing_slicing)] // TODO(#1668) Clippy exceptions need docs or fixing.
-        let no_exponent_str = &no_sign_str[..exponent_index];
+        let no_exponent_str = match no_sign_str.get(..exponent_index) {
+            Some(slice) => slice,
+            None => unreachable!("`exponent_index` must be within range."),
```

Comment:
"must be within range" or "is definitely within range because ..."? This just panics with a different message, but you should show that `exponent_index` < `no_sign_str.len()` always holds.

Reply:
Then just say something like

```
        #[allow(clippy::indexing_slicing)] // exponent_index comes from enumerate
        let no_exponent_str = &no_sign_str[..exponent_index];
```

There's no advantage in doing `.get()` and `unreachable!`.

Reply:
I find the code @robertbastian proposed to be more readable.

Reply:
aha ... okay, this means that sometimes we suppress the clippy behaviour when it does not make sense :). Good to know

Reply:
Yes, but we explain each suppression